package.json

```diff
@@ -152,7 +152,7 @@
     "terser-webpack-plugin": "^5.1.1",
     "ts-jest": "^27.1.5",
     "ts-loader": "^8.0.12",
-    "typescript": "4.1.3",
+    "typescript": "^4.9.5",
```

Comment:
Why not 5+?

Reply:
That would be a breaking change no?, we can't upgrade without a major version release

Reply:
I actually don't mind version 5 and above, it's just that build test was bothersome :cry:

Reply:
TS 5 has too many TS 5 things such as the satisfies operator which would break any consumer who isn't on 5 (a lot of people)

Reply:
Actually that is 4.9...

Not sure, but 95% sure if we ever used that and our consumers were on a lesser version they would break

Reply:
I would actually advocate that we leave the TS version alone else we break backwards compatibility. We either wait until v3 or we do something like proposed in https://github.com/microsoft/TypeScript/issues/22605 (which is a lot of work but possible)

Reply:
I have reviewed all the type definitions for the 4.9.5 update, and as far as I can tell, there doesn't seem to be any significant issues that I missed.
Even if there were any, they would likely be minor and understandable for users trying to keep up with our version

Reply:
Hold on, we ship the source code to npm? With typescript as a required dependency? That doesn't sound right. That would mean that projects using JS (no TS) are not able to use recharts at all? If that's true then we should be fixing that instead of bumping TS version

Reply:
We can use it without using TS.

Reply:
We ship the type defs to typescript consumers which means if we use a feature incompatible with the consumers TS version they will not be able to build their code without upgrading. Something like satisfies still exists in the d.ts files unless I'm misunderstanding how TS works

Reply:
https://stackoverflow.com/a/51670675

I agree with the answer in the SOF, but we can't really guard well against some PR in recharts using a new TS feature in the next minor release that will break consumers because type signatures changed. Satisfies is a good example (or something else becoming more strict). It only matters in the public API only sure but unfortunately current state of things has no separation

Reply:
How about adding a build test then?

Reply:
It seems sufficient to test only the success of the 4.1 TypeScript build and remove it in the 3.0 release

Reply:
Sure, that sounds good to me